sql-common/client.c

```diff
@@ -1948,50 +1960,116 @@ static int ssl_verify_server_cert(Vio *vio, const char* server_hostname, const c
    X509_check_host in the future.
   */
 
-  subject= X509_get_subject_name((X509 *) server_cert);
-  // Find the CN location in the subject
-  cn_loc= X509_NAME_get_index_by_NID(subject, NID_commonName, -1);
-  if (cn_loc < 0)
+#ifdef HAVE_X509_check_host
+  ret_validation= X509_check_host(server_cert, server_hostname, 0, 0, 0) != 1;
+#else
+  /*
+    Yassl will always return NULL for any call to 'X509_get_ext_d2i()'
+    and therefore the whole SAN block will be skipped and only 'CN'
+    will be checked.
+  */
+  sans= X509_get_ext_d2i(server_cert, NID_subject_alt_name, NULL, NULL);
+  if (sans != NULL)
   {
-    *errptr= "Failed to get CN location in the certificate subject";
-    goto error;
-  }
+    int number_of_sans= sk_GENERAL_NAME_num(sans);
+    for (i= 0; ret_validation != 0 && i < number_of_sans; ++i)
+    {
+        GENERAL_NAME *san= sk_GENERAL_NAME_value(sans, i);
+        if(san == NULL)
+        {
+          *errptr = "Failed to get item from SAN list";
+          goto error;
+        }
+        if (san->type == GEN_DNS)
```

Comment:
What if subjectAltName will be provided in form IP:127.0.0.1 (example) ? I do not see support for this.